python/cudf/cudf/core/column/decimal.py

```diff
@@ -168,16 +170,35 @@ def _binaryop(self, other: ColumnBinaryOperand, op: str):
 
         return result
 
+    def _scalar_to_plc_scalar(self, scalar: ScalarLike) -> plc.Scalar:
```

Comment:
Now that #17422 is merged I think we can stop special-casing this and see if anything breaks. WDYT? It does mean that decimal conversions in tests will fail if run with an older version of pyarrow, but I think that's an OK tradeoff. We might have to put some conditional xfails into our test suite for the "oldest" test runs.

Reply:
I opened https://github.com/rapidsai/cudf/pull/18035 to dedicate to avoid the decimal special casing.

Can discuss on that PR, but IIUC, to avoid these conversion on the Python side, we would need pyarrow APIs introduced in pyarrow 19

Reply:
Good call. Let's discuss there, I responded in https://github.com/rapidsai/cudf/pull/18035#issuecomment-2679256812